x-pack/plugins/ml/public/application/components/scatterplot_matrix/scatterplot_matrix.tsx

```diff
@@ -442,6 +522,28 @@ export const ScatterplotMatrix: FC<ScatterplotMatrixProps> = ({
                 </EuiFormRow>
               </EuiFlexItem>
             )}
+            {splom ? (
+              <EuiFlexItem grow={false}>
+                <EuiLink
+                  onClick={async () => {
+                    const customVisLink = getCustomVisualizationLink();
+                    await application.navigateToApp('visualize#', {
```

Comment:
Do we want it opening in a new tab or the same tab? Currently after the Visualize tab opens, the Save and return / Cancel buttons open up Dashboard. If it opens in the same tab, can the Save and return / Cancel buttons return you to the DFA page from which Visualize was opened?

Reply:
Opening it up in a new tab in e14f32ff3f779436910ec8787c521251916a7972

Reply:
This happens only if you add the charts to a dashboard from Visualize. Looks like that action adds this to session storage:

```
EMBEDDABLE_STATE_TRANSFER: {
    "visualize": {
        "originatingApp": "dashboards",
        "embeddableId": "e14aa766-2fc1-40c1-ac2f-8daeb9fa4c0c",
        "searchSessionId": "d95d87ba-8eb0-4034-b23e-57423357f4e2"
    }
}
```

Any subsequent opening of `visualize` from the ML DFA pages will then result in the 'Save and return' and 'Cancel' buttons appearing in the header and redirecting to the dashboard rather than ML. Maybe we can set ML as the originating app, or if not, clear that key? Makes sense to address in a follow up.